File: study/ts/script/data/ActivityResourceBackItemData.ts
import { BaseData } from './BaseData';
import { G_ConfigLoader } from '../init';
import { ConfigNameConst } from '../const/ConfigNameConst';
let schema = {};
schema['id'] = [
    'number',
    0
];
schema['awards'] = [
    'object',
    {}
];
schema['state'] = [
    'number',
    0
];
schema['value'] = [
    'number',
    0
];
schema['describle'] = [
    'string',
    ''
];
schema['gold'] = [
    'number',
    0
];
schema['coin'] = [
    'number',
    0
];
schema['percent'] = [
    'number',
    0
];
export interface ActivityResourceBackItemData {
getId(): number
setId(value: number): void
getLastId(): number
getAwards(): Object
setAwards(value: Object): void
getLastAwards(): Object
getState(): number
setState(value: number): void
getLastState(): number
getValue(): number
setValue(value: number): void
getLastValue(): number
getDescrible(): string
setDescrible(value: string): void
getLastDescrible(): string
getGold(): number
setGold(value: number): void
getLastGold(): number
getCoin(): number
setCoin(value: number): void
getLastCoin(): number
getPercent(): number
setPercent(value: number): void
getLastPercent(): number
}
export class ActivityResourceBackItemData extends BaseData {

    public static schema = schema;

    public clear () {
    }
    public reset () {
    }
    public updateData (message) {
        let id = message['id'];
        if (id) {
            this.setId(id);
        }
        let awards = message['awards'];
        if (awards) {
            this.setAwards(awards);
        }
        let state = message['state'];
        if (state) {
            this.setState(message.state || 0);
        }
        let value = message['value'];
        if (value) {
            this.setValue(value);
        }
        let ResourceRecoveryConfig = G_ConfigLoader.getConfig(ConfigNameConst.RESOURCE_RECOVERY);
        let config = ResourceRecoveryConfig.get(id || 0);
        console.assert(config != null, 'can not find resource_recovery id = ' + (id || 0));
        if (config) {
            this.setCoin(config.coin_price);
            this.setGold(config.acer_price);
            this.setPercent(config.percent / 100);
            this.setDescrible(config.name);
        }
    }
    public isAlreadyBuy () {
        return this.getState() == 1;
    }
}

File: study/lua/src/app/data/ActivityResourceBackItemData.lua
-- Author: nieming
-- Date:2018-02-18 11:17:36
-- Describle：

local BaseData = require("app.data.BaseData")
local ActivityResourceBackItemData = class("ActivityResourceBackItemData", BaseData)
local ResourceRecoveryConfig = require("app.config.resource_recovery")

local schema = {}
--schema
schema["id"] = {"number", 0}
schema["awards"] = {"table", {}}
schema["state"] = {"number", 0}
schema["value"] = {"number", 0}
schema["describle"] = {"string", ""}
schema["gold"] = {"number", 0}   --元宝  完美找回
schema["coin"] = {"number", 0} -- 硬币  普通找回
schema["percent"] = {"number", 0} -- 普通 资源百分百
ActivityResourceBackItemData.schema = schema


function ActivityResourceBackItemData:ctor(properties)
	ActivityResourceBackItemData.super.ctor(self, properties)

end

function ActivityResourceBackItemData:clear()

end

function ActivityResourceBackItemData:reset()

end

function ActivityResourceBackItemData:updateData(message)
	local id = rawget(message, "id")
	if id then
		self:setId(id)
	end

	local awards = rawget(message, "awards")
	if awards then
		self:setAwards(awards)
	end

	local state = rawget(message, "state")
	if state then
		self:setState(message.state or 0)
	end

	local value = rawget(message, "value")
	if value then
		self:setValue(value)
	end

	local config = ResourceRecoveryConfig.get(id or 0)
	assert(config ~= nil, "can not find resource_recovery id = "..(id or 0))
	if config then
		self:setCoin(config.coin_price)
		self:setGold(config.acer_price)
		self:setPercent(config.percent/100.0)
		self:setDescrible(config.name)
	end
end



function ActivityResourceBackItemData:isAlreadyBuy()
	return self:getState() == 1
end





return ActivityResourceBackItemData
